modulus/distributed/mappings.py

```diff
@@ -77,48 +85,23 @@ class _GatherFromParallelRegion(torch.autograd.Function):
     """Gather the input from parallel region and concatenate."""
 
     @staticmethod
-    def symbolic(graph, input_, dim_, group_):  # pragma: no cover
-        return _gather(input_, dim_, group=DistributedManager().group(group_))
+    def symbolic(graph, input_, dim_, group_, shapes_):  # pragma: no cover
+        return _gather(input_, dim_, shapes_, group=DistributedManager().group(group_))
 
     @staticmethod
-    def forward(ctx, input_, dim_, group_):  # pragma: no cover
+    def forward(ctx, input_, dim_, shapes_, group_):  # pragma: no cover
         ctx.dim = dim_
         ctx.group = group_
-        return _gather(input_, dim_, group=DistributedManager().group(group_))
+        return _gather(
+            input_, dim_, shapes_=shapes_, group=DistributedManager().group(group_)
+        )
 
     @staticmethod
     def backward(ctx, grad_output):  # pragma: no cover
         return (
             _split(grad_output, ctx.dim, group=DistributedManager().group(ctx.group)),
             None,
             None,
-        )
-
-
-class _GatherWithinParallelRegion(torch.autograd.Function):
```

Comment:
Is this not needed anymore because of the shared params reduction hook?

Reply:
It is not used anywhere inside the code so I deleted it.